src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (!victim.hidden && victim instanceof Herbivore) {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+    }
```

Comment:
```suggestion
    if (victim.health <= 0) {
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
    }
```
I recommend making it less or equal to zero. Because if you'd subtract, for example, 23 health for every bite, an animal wouldn't have 0 health. But if animal has health below 0 it's considered dead as well